src/WingmanDataTrail/MetricVizPanel/UsageData.tsx

```diff
@@ -0,0 +1,129 @@
+import { css, cx } from '@emotion/css';
+import { type GrafanaTheme2, type IconName } from '@grafana/data';
+import { Button, Dropdown, Icon, Menu, Tooltip, useStyles2 } from '@grafana/ui';
+import React from 'react';
+
+import { type MetricUsageType } from 'WingmanDataTrail/ListControls/MetricsSorter/MetricUsageFetcher';
+
+import { type WithUsageDataPreviewPanelState } from './WithUsageDataPreviewPanel';
+
+// TODO: pass full MetricUsageDetailsObject into UsageSectionProps?
+export interface UsageSectionProps {
+  usageType: MetricUsageType;
+  usageCount: number;
+  singularUsageType: string;
+  pluralUsageType: string;
+  icon: IconName;
+  dashboardItems: WithUsageDataPreviewPanelState['dashboardItems'];
+}
+
+export function UsageData({
+  usageType,
+  usageCount,
+  singularUsageType,
+  pluralUsageType,
+  icon,
+  dashboardItems,
+}: Readonly<UsageSectionProps>) {
+  const styles = useStyles2(getStyles);
+
+  return (
+    <div className={styles.usageContainer} data-testid="usage-data-panel">
+      {usageType === 'dashboard-usage' ? (
+        <>
+          <Dropdown
+            placement="right-start"
+            overlay={
+              <Menu style={{ maxWidth: '240px', maxHeight: '245px', overflowY: 'auto' }}>
+                {dashboardItems.map((item) => (
+                  <Menu.Item
+                    key={item.id}
+                    label=""
+                    url={item.url}
+                    target="_blank"
+                    className={styles.menuItem}
+                    component={() => (
+                      <Tooltip
+                        content={`Found ${item.count} ${item.count === 1 ? 'time' : 'times'} in ${item.label}`}
```

Comment:
nit: we could reinforce the "usage" idea:

```suggestion
                        content={`Used ${item.count} ${item.count === 1 ? 'time' : 'times'} in ${item.label}`}
```